src/components/LibraryContainer.tsx

```diff
@@ -20,12 +20,12 @@ export interface LibraryContainerProps {
 }
 
 export interface LibraryContainerStates {
-    inSearchMode: boolean,
-    searchText: string,
-    selectedCategories: string[],
-    structured: boolean,
-    detailed: boolean
-    showItemSummary: boolean
+    inSearchMode?: boolean,
```

Comment:
@pboyer can you splain this pattern to me? 
If all the properties of these interfaces are optional, then shouldn't the code also reflect that and be fault tolerant to  missing properties? Is it?